test/raxx/router_test.exs

```diff
@@ -60,87 +60,263 @@ defmodule Raxx.RouterTest do
+      {[response], _state} = SectionRouter.handle_head(request, %{authorization: :pass})
+      assert "Users page" == response.body
+    end
+
+    test "will route to variable segment path" do
+      request = Raxx.request(:GET, "/users/34")
+      {[response], _state} = SectionRouter.handle_head(request, %{authorization: :pass})
+      assert "User page 34" == response.body
+    end
+
+    test "will route on method" do
+      request = Raxx.request(:POST, "/users")
+      {[response], _state} = SectionRouter.handle_head(request, %{authorization: :pass})
+      assert "User created " == response.body
+    end
+
+    test "will forward whole request to controller" do
+      request =
+        Raxx.request(:POST, "/users")
+        |> Raxx.set_body(true)
+
+      {[], state} = SectionRouter.handle_head(request, %{authorization: :pass})
+      {[], state} = SectionRouter.handle_data("Bob", state)
+      {[response], _state} = SectionRouter.handle_tail([], state)
+      assert "User created Bob" == response.body
+    end
+
+    test "will route on catch all" do
+      request = Raxx.request(:GET, "/random")
+      {[response], _state} = SectionRouter.handle_head(request, %{authorization: :pass})
+      assert "Not found" == response.body
+    end
+
+    @tag :skip
+    test "adds the action module to logger metadata" do
+      request = Raxx.request(:GET, "/")
+      _ = SectionRouter.handle_head(request, %{authorization: :pass})
+      metadata = Logger.metadata()
+      assert "Raxx.RouterTest.HomePage" = Keyword.get(metadata, :"raxx.action")
+      assert "%{method: :GET, path: []}" = Keyword.get(metadata, :"raxx.route")
+    end
+
+    test "will raise return error if fails to route simple request" do
+      request = Raxx.request(:GET, "/invalid")
+
+      assert_raise ReturnError, fn ->
+        SectionRouter.handle_head(request, %{authorization: :pass})
+      end
+    end
+
+    test "will raise return error if fails to route streamed request" do
+      request =
+        Raxx.request(:POST, "/invalid")
+        |> Raxx.set_body(true)
+
+      {[], state} = SectionRouter.handle_head(request, %{authorization: :pass})
+      {[], state} = SectionRouter.handle_data("Bob", state)
+
+      assert_raise ReturnError, fn ->
+        SectionRouter.handle_tail([], state)
+      end
+    end
+
+    test "middleware as list is applied" do
+      request = Raxx.request(:GET, "/")
+      {[response], _state} = SectionRouter.handle_head(request, %{authorization: :pass})
+      assert "compile-time" == Raxx.get_header(response, "x-test")
+    end
 
-    {[], state} = MyRouter.handle_head(request, :state)
-    {[], state} = MyRouter.handle_data("Bob", state)
+    test "middleware as function is applied" do
+      request = Raxx.request(:GET, "/users")
+      {[response], _state} = SectionRouter.handle_head(request, %{authorization: :pass})
+      assert 200 == response.status
+      assert "run-time" == Raxx.get_header(response, "x-test")
```

Comment:
This doesn't prove that the config function was called at runtime, maybe the runtime configuration function could send a message that the test could receive or something

Reply:
I wasn't really trying to prove anything It was just a name to separate from the test before

Reply:
Sorry, I phrased that wrong. I meant that demonstrating that the function gets called at the right time at runtime would be a good addition to this test.

Reply:
Fair enough,

last commit sufficient?

Reply:
no probs. Will release 0.17.3 now